src/cls/rgw/cls_rgw.cc

```diff
@@ -3266,10 +3266,13 @@ static int rgw_cls_gc_list(cls_method_context_t hctx, bufferlist *in, bufferlist
   }
 
   cls_rgw_gc_list_ret op_ret;
-  int ret = gc_list_entries(hctx, op.marker, op.max, op.expired_only, op_ret.entries, &op_ret.truncated);
+  string next_marker;
+  int ret = gc_list_entries(hctx, op.marker, op.max, op.expired_only, 
+   op_ret.entries, &op_ret.truncated, next_marker);
   if (ret < 0)
     return ret;
 
+  op_ret.next_marker = next_marker;
```

Comment:
it doesn't look like the local variable `next_marker` is needed here, you can just pass `op_ret.next_marker` to `gc_list_entries()`